src/engine/sparqlExpressions/SparqlExpressionValueGetters.cpp

```diff
@@ -315,6 +318,39 @@ OptIri IriValueGetter::operator()(
   }
 }
 
+// _____________________________________________________________________________
+UnitOfMeasurement UnitOfMeasurementValueGetter::operator()(
+    ValueId id, const EvaluationContext* context) const {
+  // Use cache to remember fully parsed units for reoccurring ValueIds
+  return cache_.getOrCompute(
+      id, [&context](const ValueId& value) -> UnitOfMeasurement {
+        // Get string content of ValueId
+        auto str = ExportQueryExecutionTrees::idToLiteralOrIri(
+            context->_qec.getIndex(), value, context->_localVocab);
+        // Use LiteralOrIri overload for actual computation
+        if (str.has_value()) {
+          return UnitOfMeasurementValueGetter{}(str.value(), context);
+        }
+        return UnitOfMeasurement::UNKNOWN;
```

Comment:
If you implement my suggestion of skipping `int` values directly in the `idToLiteralOrIri` function, then this line becomes testable.